src/webportal/src/app/job/job-view/fabric/JobList/TopBar.jsx

```diff
@@ -63,23 +64,45 @@ function KeywordSearchBox() {
 
 function TopBar() {
   const [active, setActive] = useState(true);
-  const {allJobs, refreshJobs, selectedJobs, stopJob, username, filter, setFilter} = useContext(Context);
+  const {refreshJobs, selectedJobs, stopJob, username, filter, setFilter} = useContext(Context);
+
+  const allStatuses = ['Waiting', 'Succeeded', 'Running', 'Stopped', 'Failed'];
 
   const {users, virtualClusters, statuses} = useMemo(() => {
     const users = Object.create(null);
     const virtualClusters = Object.create(null);
     const statuses = Object.create(null);
 
-    if (allJobs !== null) {
-      allJobs.forEach(function(job) {
-        users[job.username] = true;
-        virtualClusters[job.virtualCluster] = true;
-        statuses[getStatusText(job)] = true;
+    fetch(`${webportalConfig.restServerUri}/api/v1/user`)
+      .then((response) => {
+        return response.json();
+      }).then((body) => {
+        body.forEach((userBody) => {
+          users[userBody['username']] = true;
+        });
+      }).catch((err) => {
+        alert(err.message);
       });
-    }
+
+    fetch(`${webportalConfig.restServerUri}/api/v1/virtual-clusters`)
+      .then((response) => {
+        return response.json();
+      }).then((body) => {
+        for (let virtualCluster in body) {
```

Comment:
```suggestion
        for (const virtualCluster of Object.keys(body)) {
```

Without `hasOwnProperty`